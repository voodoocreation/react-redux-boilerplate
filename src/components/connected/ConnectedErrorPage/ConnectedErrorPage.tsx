import * as React from "react";
import { connect } from "react-redux";

import { IError } from "../../../models/error.models";
import { IStoreState } from "../../../reducers/root.reducers";
import * as selectors from "../../../selectors/root.selectors";

import ErrorPage from "../../presentation/ErrorPage/ErrorPage";

interface IProps {
  error: IError | undefined;
}

class ConnectedErrorPage extends React.Component<IProps> {
  public render() {
    return <ErrorPage {...this.props.error} />;
  }
}

const mapStateToProps = (state: IStoreState) => ({
  error: selectors.getAppError(state)
});

const ConnectedErrorPageWrapped = connect(mapStateToProps)(ConnectedErrorPage);

export default ConnectedErrorPageWrapped;
