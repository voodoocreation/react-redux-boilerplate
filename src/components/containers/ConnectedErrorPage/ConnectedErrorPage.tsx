import * as React from "react";
import { connect } from "react-redux";

import ErrorPage from "../../presentation/ErrorPage/ErrorPage";

import * as selectors from "../../../selectors/root.selectors";

interface IProps {
  error?: IError;
}

class ConnectedErrorPage extends React.Component<IProps> {
  public render() {
    return <ErrorPage {...this.props.error} />;
  }
}

const mapStateToProps = (state: any) => ({
  error: selectors.getPageError(state)
});

export default connect<IProps>(mapStateToProps)(ConnectedErrorPage);
