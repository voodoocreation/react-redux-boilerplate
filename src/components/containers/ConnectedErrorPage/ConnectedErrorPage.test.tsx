import { render } from "enzyme";
import * as React from "react";

import ConnectedErrorPage from "./ConnectedErrorPage";

jest.mock("react-redux", () => ({
  connect: () => (component: any) => component
}));

const setup = (fn: any) => {
  const props = {
    error: {
      message: "Not found",
      status: 404
    }
  };

  return {
    actual: fn(<ConnectedErrorPage {...props} />),
    props
  };
};

describe("<ConnectedErrorPage />", () => {
  it("renders correctly", () => {
    const { actual } = setup(render);
    expect(actual).toMatchSnapshot();
  });
});
