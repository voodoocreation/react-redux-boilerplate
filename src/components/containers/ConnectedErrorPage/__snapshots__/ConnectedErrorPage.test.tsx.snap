// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ConnectedErrorPage /> renders correctly 1`] = `
Array [
  <h1>
    Page not found
  </h1>,
  <p>
    This page does not exist.
  </p>,
]
`;
